packages/devextreme/js/ui/context_menu/ui.context_menu.js

```diff
@@ -603,25 +605,46 @@ class ContextMenu extends MenuBase {
         }
     }
 
+    _initScrollView($container) {
+        const maxHeight = this._getMaxHeight($container);
+        const containerHeight = getOuterHeight($container);
+        const menuHeight = Math.min(containerHeight, maxHeight);
+
+        $container.css('position', 'fixed');
+        $container.css('height', menuHeight);
+        this._createComponent($container, ScrollView, {});
+    }
+
+    _getMaxHeight($element) {
+        const offsetTop = $element.offset().top;
+        const windowHeight = getOuterHeight(window);
+        const maxHeight = Math.max(offsetTop, windowHeight - offsetTop);
+
+        return Math.min(windowHeight, maxHeight);
+    }
+
     _showSubmenu($item) {
         const node = this._dataAdapter.getNodeByItem(this._getItemData($item));
 
         this._hideSubmenusOnSameLevel($item);
 
         if(!this._hasSubmenu(node)) return;
 
-        const $submenu = $item.children(`.${DX_SUBMENU_CLASS}`);
+        let $submenu = $item.children(`.${DX_SUBMENU_CLASS}`);
         const isSubmenuRendered = $submenu.length;
 
         super._showSubmenu($item);
 
         if(!isSubmenuRendered) {
             this._renderSubmenuItems(node, $item);
+            $submenu = $item.children(`.${DX_SUBMENU_CLASS}`);
         }
 
         if(!this._isSubmenuVisible($submenu)) {
             this._drawSubmenu($item);
         }
+
+        this._initScrollView($submenu);
```

Comment:
why do u pass submenu here?
Your solution still does not work.
Just click to open contextMenu and see that submenu takes windows height because offset if 0